### PR TITLE
Add User Guide Sections for Import/Export

### PR DESCRIPTION
**Pull request by dandaandaaaaaan**
## Changes
This PR updates only the User Guide and adds commands related to import and export.
A staging version of the updated User Guide can be found [here](https://dandaandaaaaaan.github.io/tp/UserGuide.html).

## Closes Issues
* #71

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
 
 ### Saving the data
 
-AddressBook data are saved in the hard disk automatically after any command that changes the data. There is no need to save manually.
+ContacX data are saved in the hard disk automatically after any command that changes the data in JSON format at `[JAR file location]/data/contacx.json`.
 
+### Exporting the data: `exportcsv`
+
+Exports the current list as a CSV file that can be imported through ContacX or other contact software such as Google/Outlook
```
Comment (sharpstorm) [inline docs/UserGuide.md:163]: I suggest being more specific for the data being stored
> list *of contacts*
```suggestion
Exports the current list as a CSV file that can be imported through ContaX or other contact software such as Google/Outlook
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
+
+Format: `exportcsv EXPORTTYPE`
+* Exports CSV as per the specified format.
+* If `EXPORTTYPE` is left empty, it will default to `EXPORTTYPE = 1` and export in custom ContacX format
```
Comment (sharpstorm) [inline docs/UserGuide.md:167]: If it can be left empty, then it should be surrounded by square brackets.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
+   * `EXPORTTYPE=2` for Google Contacts format
+   * `EXPORTTYPE=3` for Microsoft Outlook format
+* File will be saved on the directory `[JAR file location]/data/addressbook.csv`
+
+Examples:
+* `exportcsv 2`: Exports the current address book as a CSV file ready to be uploaded onto Google Contacts 
 ### Editing the data file
 
-AddressBook data are saved as a JSON file `[JAR file location]/data/addressbook.json`. Advanced users are welcome to update data directly by editing that data file.
+ContacX data are saved as a JSON file `[JAR file location]/data/contacx.json`. Advanced users are welcome to update data directly by editing that data file.
```
Comment (sharpstorm) [inline docs/UserGuide.md:178]: Spelling issue. It is also present in multiple other locations.
```suggestion
ContaX data are saved as a JSON file `[JAR file location]/data/contax.json`. Advanced users are welcome to update data directly by editing that data file.
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
 
 <div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
-If your changes to the data file makes its format invalid, AddressBook will discard all data and start with an empty data file at the next run.
+If your changes to the data file makes its format invalid, ContacX will discard all data and start with an empty data file at the next run.
```
Comment (sharpstorm) [inline docs/UserGuide.md:181]: Spelling Issue.
```suggestion
If your changes to the data file makes its format invalid, ContaX will discard all data and start with an empty data file at the next run.
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
 </div>
 
+### Importing data
+
+In order to import data from external sources into ContacX, there are the following options:
+
+* Replacing `contacx.json` file with a valid ContacX formatted generated by another system with ContacX
+* Simple CSV import (GUI-based option)
+* Advanced CSV import (Command based)
```
Comment (sharpstorm) [inline docs/UserGuide.md:190]: I suggest adding hyperlinks here.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
+
+### Simple CSV Import
+
+Select the `Import` button under `File->Import` on the top bar.
+
+!['basic import UI'](images/BasicImportUI.png)
+
+Upon click, the default OS file picker should show up, with filters for only .csv and .json files to be accepted. Accessing import via the top menu will allow users to import their task and appointments list as JSON/CSV files that were generated by another instance of ContacX.
```
Comment (sharpstorm) [inline docs/UserGuide.md:198]: I think this is too specific for a user-facing guide. Consider simplification to
> Upon ~click~ *clicking*, you will be prompted to select `.csv` or `.json` files.

File: docs/UserGuide.md [hunk 7]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
+
+### Advanced Data Import : `importcsv`
+
+Format: `importcsv n/COLUMNNUM p/COLUMN_PERSON e/COLUMN_EMAIL a/COLUMN_ADDRESS t/COLUMN_TAGS`
+
+* Each argument represents the corresponding attribute for a person. n = Name, p = Phone, e = Email, a = Address, t = Tagged
```
Comment (sharpstorm) [inline docs/UserGuide.md:204]: I suggest using a 2nd level bullet for each of the arguments.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
+* If any argument is omitted, it will follow the next available column based on the default sequence of
+  Name - Phone - Email - Address - Tagged
+* Tutorials will have templates for importing for the following contact csv formats:
```
Comment (sharpstorm) [inline docs/UserGuide.md:207]: What is tutorial referring to?

File: docs/UserGuide.md [hunk 9]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
+!['basic import UI'](images/BasicImportUI.png)
+
+Upon click, the default OS file picker should show up, with filters for only .csv and .json files to be accepted. Accessing import via the top menu will allow users to import their task and appointments list as JSON/CSV files that were generated by another instance of ContacX.
+
+### Advanced Data Import : `importcsv`
+
+Format: `importcsv n/COLUMNNUM p/COLUMN_PERSON e/COLUMN_EMAIL a/COLUMN_ADDRESS t/COLUMN_TAGS`
+
+* Each argument represents the corresponding attribute for a person. n = Name, p = Phone, e = Email, a = Address, t = Tagged
+* If any argument is omitted, it will follow the next available column based on the default sequence of
```
Comment (sharpstorm) [inline docs/UserGuide.md:205]: Arguments that can be omitted should be placed in square brackets.

File: docs/UserGuide.md [hunk 10]
```diff
@@ -156,16 +156,65 @@ Format: `exit`
...
+  Name - Phone - Email - Address - Tagged
+* Tutorials will have templates for importing for the following contact csv formats:
+  * Google Contacts
+  * Outlook Contacts
+
+Examples:
+* `importCSV n/2 p/3 e/5 a/6 t/4`
+  * Reads from a CSV file treating the 2nd row as names, 3rd row as phone numbers, 5th row as email addresses, 4th row as tagged.
+* `importCSV n/2`
+  * Reads from CSV treating the 2nd row as names, 3rd row as phone numbers, 4th row as email address, 5th row as tagged.
```
Comment (sharpstorm) [inline docs/UserGuide.md:215]: Command capitalization is not consistent. Please choose a single way of capitalizing `csv`

File: docs/UserGuide.md [hunk 11]
```diff
@@ -190,3 +239,5 @@ Action | Format, Examples
 **Find** | `find KEYWORD [MORE_KEYWORDS]`<br> e.g., `find James Jake`
 **List** | `list`
 **Help** | `help`
+**ExportCSV** | `exportCSV EXPORTTYPE`
+**ImportCSV** | `importCSV n/COLUMNNUM p/COLUMN_PERSON e/COLUMN_EMAIL a/COLUMN_ADDRESS t/COLUMN_TAGS`
```
Comment (sharpstorm) [inline docs/UserGuide.md:243]: Consider adding example commands.

**Review by sharpstorm — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: amended to fit suggestion, thanks!

File: docs/UserGuide.md (hunk 2 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: thanks for the catch, didn't notice it when doing

File: docs/UserGuide.md (hunk 3 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: thanks for catching will stop renaming stuff wrongly

File: docs/UserGuide.md (hunk 4 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: thanks for catching

File: docs/UserGuide.md (hunk 5 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: thanks for suggestion i've implemented

File: docs/UserGuide.md (hunk 7 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: thanks for the suggestion i've implemented it

File: docs/UserGuide.md (hunk 8 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: updated for clarity. I intend to put the command templates here into this user guide later on once the feature is more developed

File: docs/UserGuide.md (hunk 9 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md:210]: thanks for catch, resolved

File: docs/UserGuide.md (hunk 11 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: thanks for the suggestion i added a few

File: docs/UserGuide.md (hunk 10 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: thanks for the catch accidentlaly named this wrongly

File: docs/UserGuide.md (hunk 6 above)
Comment (dandaandaaaaaan) [reply docs/UserGuide.md]: ok will fix this

**Review by sharpstorm — APPROVED** [review]
lgtm.

**Review by Wei-Xinn — APPROVED** [review]
lgtm!

**Pull request merged**